### ISSUE TITLE
TestAttachAPI test failed with com.sun.tools.attach.AgentLoadException: ATTACH_ERR AgentLoadException

### ISSUE THREAD

**Issue by llxia** [eclipse/openj9#238]
Link to Jenkins console output: https://ci.eclipse.org/openj9/view/Test/job/Test-Extended-linux_390-64_cmprssptrs/16/consoleFull

Output:
```
02:40:10 ===============================================
02:40:10 Running test TestAttachAPI_SE90_0 ...
02:40:10 ===============================================
02:40:10 test with NoOptions
02:40:10 { "/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/jdk/bin/java"  -Xcompressedrefs  \
02:40:10 --add-exports=java.base/com.ibm.tools.attach.target=ALL-UNNAMED \
02:40:10 -cp "/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/../../jvmtest/TestConfig/resources:/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/../../jvmtest/TestConfig/lib/testng.jar:/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/../../jvmtest/TestConfig/lib/jcommander.jar:/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/../../jvmtest/Java8andUp/GeneralTest.jar" \
02:40:10 -Djava.sidecar="--add-exports=java.base/com.ibm.tools.attach.target=ALL-UNNAMED" \
02:40:10 -Dcom.ibm.tools.attach.enable=yes \
02:40:10 -Dcom.ibm.tools.attach.timeout=15000 \
02:40:10 org.testng.TestNG -d "/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/test_output_1507099002/TestAttachAPI_SE90_0" "/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/../../jvmtest/Java8andUp/testng.xml" -testnames TestAttachAPI \
02:40:10 -groups level.extended \
02:40:10 -excludegroups d.*.linux_390-64_cmprssptrs,d.*.arch.390,d.*.os.linux,d.*.bits.64,d.*.generic-all; \
02:40:10 if [ $? -eq 0 ] ; then echo "TestAttachAPI_SE90_0""_PASSED"; else echo "TestAttachAPI_SE90_0""_FAILED"; fi; } 2>&1 | tee -a TestTargetResult
02:40:13 [XmlSuite] [WARN] 'parallel' value 'false' is deprecated, default value will be used instead: 'none'.
02:40:13 [IncludeExcludeTestAnnotationTransformer] [INFO] exclude file is /home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/jvmtest/TestConfig/resources/excludes/default_exclude.txt
02:40:13 ...
02:40:13 ... TestNG 6.10 by Cédric Beust (cedric@beust.com)
02:40:13 ...
02:40:13 
02:40:14 [TestNG] Running:
02:40:14   /home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/jvmtest/Java8andUp/testng.xml
02:40:14 
02:40:33 [AttachApiTest] [ERROR] com.sun.tools.attach.AgentLoadException: ATTACH_ERR AgentLoadException jvmtitest:
02:40:33 	at com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.parseResponse(jdk.attach@9-internal/OpenJ9VirtualMachine.java:327)
02:40:33 	at com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.loadAgentLibrary(jdk.attach@9-internal/OpenJ9VirtualMachine.java:244)
02:40:33 	at com.sun.tools.attach.VirtualMachine.loadAgentLibrary(jdk.attach@9-internal/VirtualMachine.java:377)
02:40:33 	at org.openj9.test.attachAPI.TestAttachAPI.test_agntld01(TestAttachAPI.java:136)
02:40:33 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(java.base@9-internal/Native Method)
02:40:33 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(java.base@9-internal/NativeMethodAccessorImpl.java:62)
02:40:33 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@9-internal/DelegatingMethodAccessorImpl.java:43)
02:40:33 	at java.lang.reflect.Method.invoke(java.base@9-internal/Method.java:564)
02:40:33 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
02:40:33 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
02:40:33 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
02:40:33 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
02:40:33 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
02:40:33 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
02:40:33 	at org.testng.TestRunner.privateRun(TestRunner.java:756)
02:40:33 	at org.testng.TestRunner.run(TestRunner.java:610)
02:40:33 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
02:40:33 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
02:40:33 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
02:40:33 	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
02:40:33 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
02:40:33 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
02:40:33 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
02:40:33 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
02:40:33 	at org.testng.TestNG.runSuites(TestNG.java:1133)
02:40:33 	at org.testng.TestNG.run(TestNG.java:1104)
02:40:33 	at org.testng.TestNG.privateMain(TestNG.java:1434)
02:40:33 	at org.testng.TestNG.main(TestNG.java:1403)
02:40:33 
```

test_agntld01, test_agntld02 and test_agntld03 failed with:
```java.lang.AssertionError: missing Agent_OnAttach string```

**Comment by llxia** [comment]
Going to exclude this test for now.

**Comment by pdbain-ibm** [comment]
Is there an Issue for this?

**Comment by smlambert** [comment]
#238 the issue you are editing, is the issue for this test failure.

**Comment by pdbain-ibm** [comment]
https://github.com/eclipse/openj9/pull/236 should fix this.

**Comment by llxia** [comment]
testAgentLoading failure is similar. Going to exclude the test as well.

Output:
```
02:51:32 ===============================================
02:51:32 Running test TestSunAttachClasses_SE90_0 ...
02:51:32 ===============================================
02:51:32 test with NoOptions
02:51:32 { "/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/jdk/bin/java"  -Xcompressedrefs  \
02:51:32 --add-exports=java.base/com.ibm.tools.attach.target=ALL-UNNAMED \
02:51:32 -cp "/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/../../jvmtest/TestConfig/resources:/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/../../jvmtest/TestConfig/lib/testng.jar:/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/../../jvmtest/TestConfig/lib/jcommander.jar:/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/../../jvmtest/Java8andUp/GeneralTest.jar" \
02:51:32 -Dcom.ibm.tools.attach.enable=yes \
02:51:32 -Dcom.ibm.tools.attach.timeout=15000 \
02:51:32 -Djava.sidecar="--add-exports=java.base/com.ibm.tools.attach.target=ALL-UNNAMED" \
02:51:32 org.testng.TestNG -d "/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/test_output_1507099002/TestSunAttachClasses_SE90_0" "/home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/test/TestConfig/../../jvmtest/Java8andUp/testng.xml" -testnames TestSunAttachClasses \
02:51:32 -groups level.extended \
02:51:32 -excludegroups d.*.linux_390-64_cmprssptrs,d.*.arch.390,d.*.os.linux,d.*.bits.64,d.*.generic-all; \
02:51:32 if [ $? -eq 0 ] ; then echo "TestSunAttachClasses_SE90_0""_PASSED"; else echo "TestSunAttachClasses_SE90_0""_FAILED"; fi; } 2>&1 | tee -a TestTargetResult
02:51:37 [XmlSuite] [WARN] 'parallel' value 'false' is deprecated, default value will be used instead: 'none'.
02:51:37 [IncludeExcludeTestAnnotationTransformer] [INFO] exclude file is /home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/jvmtest/TestConfig/resources/excludes/default_exclude.txt
02:51:37 ...
02:51:37 ... TestNG 6.10 by Cédric Beust (cedric@beust.com)
02:51:37 ...
02:51:37 
02:51:37 [TestNG] Running:
02:51:37   /home/jenkins/jenkins-agent/workspace/Test-Extended-linux_390-64_cmprssptrs/jvmtest/Java8andUp/testng.xml
02:51:37 
02:51:50 PASSED: testAttachToOtherProcess
02:51:50 PASSED: testAttachToSelf
02:51:50 FAILED: testAgentLoading
02:51:50 java.lang.AssertionError: missing Agent_OnAttach string
02:51:50 	at org.testng.AssertJUnit.fail(AssertJUnit.java:59)
02:51:50 	at org.testng.AssertJUnit.assertTrue(AssertJUnit.java:24)
02:51:50 	at org.openj9.test.attachAPI.TestSunAttachClasses.testAgentLoading(TestSunAttachClasses.java:135)
02:51:50 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(java.base@9-internal/Native Method)
02:51:50 	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(java.base@9-internal/NativeMethodAccessorImpl.java:62)
02:51:50 	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@9-internal/DelegatingMethodAccessorImpl.java:43)
02:51:50 	at java.lang.reflect.Method.invoke(java.base@9-internal/Method.java:564)
02:51:50 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
02:51:50 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
02:51:50 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
02:51:50 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
02:51:50 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
02:51:50 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
02:51:50 	at org.testng.TestRunner.privateRun(TestRunner.java:756)
02:51:50 	at org.testng.TestRunner.run(TestRunner.java:610)
02:51:50 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
02:51:50 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
02:51:50 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
02:51:50 	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
02:51:50 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
02:51:50 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
02:51:50 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1293)
02:51:50 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1218)
02:51:50 	at org.testng.TestNG.runSuites(TestNG.java:1133)
02:51:50 	at org.testng.TestNG.run(TestNG.java:1104)
02:51:50 	at org.testng.TestNG.privateMain(TestNG.java:1434)
02:51:50 	at org.testng.TestNG.main(TestNG.java:1403)
```

**Comment by pdbain-ibm** [comment]
@llxia please assign this to me.  Thank you.

**Comment by DanHeidinga** [comment]
@pdbain-ibm What's the current status on this item?

**Comment by pshipton** [comment]
I believe its blocked by #252 - libjvmtitest.so is missing

**Comment by pshipton** [comment]
Fixed by #1901
